Add docs for VisibilityStateEntry

chrisdavidmills opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description

Chrome 115 adds the `VisibilityStateEntry` class, which represents a performance entry of type “visibility-state”. See https://chromestatus.com/feature/5683502144028672.

This PR adds documentation for the new class, and updates surrounding Performance API docs to mention it as appropriate.

See also my [research doc](https://docs.google.com/document/d/1nIsWNlAdkwvzlqcNJ0aC0FTAxN8IDcYmYO42kUwhGvM/edit) for more information on exactly what work has been done.

<!-- ✍️ Summarize your changes in one or two sentences -->

### Motivation

<!-- ❓ Why are you making these changes and how do they help readers? -->

### Additional details

<!-- 🔗 Link to release notes, vendor docs, bug trackers, source control, or other places providing more context -->

### Related issues and pull requests

<!-- 🔨 If this fully resolves a GitHub issue, use "Fixes #123" -->
<!-- 👉 Highlight related pull requests using "Relates to #123" -->
<!-- ❗ If another pull request should be merged first, use "**Depends on:** #123" -->


<!-- 👷‍♀️ After submitting, go to the "Checks" tab of your PR for the build status -->

chrisdavidmills committed:
Add docs for VisibilityStateEntry

chrisdavidmills committed:
Add link to Page Visibility API

noamr commented on files/en-us/web/api/visibilitystateentry/index.md:
```diff
@@ -0,0 +1,83 @@
+{{APIRef("Performance API")}}{{seecompattable}}
+
+The **`VisibilityStateEntry`** interface provides timings of page visibility state changes, i.e., when a tab changes from the foreground to the background or vice versa.
+
```
An important bit that I'm missing here is that the document would always have at least one `visibility-state` entry at time 0 - which would represent the visiblity state when the document was created.

Mawajingga replied:
Delete

noamr commented on files/en-us/web/api/visibilitystateentry/index.md:
```diff
@@ -0,0 +1,83 @@
+## Specifications
+
+{{Specifications}}
+
```
Would be good to have a link to the spec:
https://html.spec.whatwg.org/multipage/interaction.html#visibilitystateentry

Mawajingga replied:
Delete

chrisdavidmills committed:
Make information about initial visbility-state entry more prominent

chrisdavidmills committed:
Merge branch 'main' into add-visibilitystateentry

noamr reviewed (commented):
LGTM

chrisdavidmills committed:
Merge branch 'main' into add-visibilitystateentry

noamr commented:
> LGTM

(I mean a non-authoritative LGTM from a technical point of view)

chrisdavidmills commented:
> > LGTM
> 
> (I mean a non-authoritative LGTM from a technical point of view)

Yup, that's perfect. Thx for taking the time to review!

Elchi3 reviewed (commented):
Could you also update https://developer.mozilla.org/en-US/docs/Web/API/performance_property to say it is Window-context only?

Also, I wrote a guide about "Performance data" a while ago and it would be nice to add this to it: https://developer.mozilla.org/en-US/docs/Web/API/Performance_API/Performance_data 
There is a section on this page about buffer sizes and my source is https://w3c.github.io/timing-entrytypes-registry/#registry. However, I don't see `visibility-state` performance entries in that registry. Maybe @noamr knows more.

Elchi3 commented:
Also, I wonder if we should integrate this into https://developer.mozilla.org/en-US/docs/Web/API/Performance_API/Navigation_timing

Elchi3 reviewed (commented):
Also, updates to PerformanceEntry would be good:

- https://developer.mozilla.org/en-US/docs/Web/API/PerformanceEntry/duration add "visibility-state" to the list "For the following entry types, duration is not applicable, and in this case the value is always 0:"
- https://developer.mozilla.org/en-US/docs/Web/API/PerformanceEntry/name add to the table
- https://developer.mozilla.org/en-US/docs/Web/API/PerformanceEntry/startTime add to "Value" section

chrisdavidmills commented:
Thanks for the review @Elchi3 . To summarize the changes in my next commit (it weirdly won't let me add replies to your individual comments):

* I have added `VisibilityStateEntry` into the table on the performance_property page
* I have added it in all appropriate places in the Performance_data page. I couldn't see a max buffer entries defined for it either, so I've entered it as `infinite` for now. But yes, maybe @noamr can help confirm this value.
* I've not added anything to the Navigation_timing page, as I'm not really sure what you were thinking of here. It doesn't really seem directly relevant to me.
* Changes to `name`, `duration`, and `startTime` — all done.

chrisdavidmills committed:
Changes made according to Elchi3 comments

noamr commented:
> Thanks for the review @Elchi3 . To summarize the changes in my next commit (it weirdly won't let me add replies to your individual comments):
> 
> * I have added `VisibilityStateEntry` into the table on the performance_property page
> * I have added it in all appropriate places in the Performance_data page. I couldn't see a max buffer entries defined for it either, so I've entered it as `infinite` for now. But yes, maybe @noamr can help confirm this value.

It's 50. I should add it to the registry spec.

> * I've not added anything to the Navigation_timing page, as I'm not really sure what you were thinking of here. It doesn't really seem directly relevant to me.
> * Changes to `name`, `duration`, and `startTime` — all done.

chrisdavidmills committed:
Updated visibility-state max buffer entry to correct value, as confirmed by noamr

chrisdavidmills commented:
> It's 50. I should add it to the registry spec.

Thanks @noamr ; page updated.

Elchi3 reviewed (approved):
Great work, thanks @chrisdavidmills 👍
